rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp

```diff
@@ -0,0 +1,416 @@
+  }
+  setStatus(rviz_common::properties::StatusProperty::Ok, "URDF", "Robot model parsed Ok");
+
+  for (std::map<std::string, urdf::JointSharedPtr>::iterator it = robot_model_->joints_.begin();
+    it != robot_model_->joints_.end(); it++)
+  {
+    urdf::JointSharedPtr joint = it->second;
+    if (joint->type == urdf::Joint::REVOLUTE || joint->type == 2) {
+      std::string joint_name = it->first;
+      urdf::JointLimitsSharedPtr limit = joint->limits;
+      joints_[joint_name] = createJoint(joint_name);
+      joints_[joint_name]->setMaxEffort(limit->effort);
+    }
+  }
+}
+
+void EffortDisplay::subscribeToRobotDescription()
+{
+  if (this->robot_description_topic_ == robot_description_property_->getStdString()) {
+    return;
+  }
+  this->robot_description_topic_ = robot_description_property_->getStdString();
+
+  using std::placeholders::_1;
+
+  try {
+    rclcpp::SubscriptionOptions sub_opts;
+    sub_opts.event_callbacks.message_lost_callback =
+      [&](rclcpp::QOSMessageLostInfo & info)
+      {
+        std::ostringstream sstm;
+        sstm << "Some messages were lost:\n>\tNumber of new lost messages: " <<
+          info.total_count_change << " \n>\tTotal number of messages lost: " <<
+          info.total_count;
+        setStatus(
+          rviz_common::properties::StatusLevel::Warn,
+          "Array Topic",
+          QString(sstm.str().c_str()));
+      };
+
+    this->subscription_ = context_->getRosNodeAbstraction().lock()->get_raw_node()->
+      template create_subscription<std_msgs::msg::String>(
+      robot_description_property_->getStdString(),
+      rclcpp::QoS(1).transient_local(),
+      std::bind(&EffortDisplay::topic_callback, this, _1),
+      sub_opts);
+    setStatus(rviz_common::properties::StatusLevel::Ok, "Array Topic", "OK");
+  } catch (rclcpp::exceptions::InvalidTopicNameError & e) {
+    setStatus(
+      rviz_common::properties::StatusLevel::Error,
+      "Array Topic",
+      QString("Error subscribing: ") + e.what());
+  }
+}
+
+void EffortDisplay::load()
+{
+  this->subscribeToRobotDescription();
+}
+
+void EffortDisplay::update(float wall_dt, float ros_dt)
+{
+  (void) wall_dt;
+  (void) ros_dt;
+}
+
+// Set the current color and alpha values for each visual.
+void EffortDisplay::updateColorAndAlpha()
+{
+  float width = width_property_->getFloat();
+  float scale = scale_property_->getFloat();
+
+  for (size_t i = 0; i < visuals_.size(); i++) {
+    visuals_[i]->setWidth(width);
+    visuals_[i]->setScale(scale);
+  }
+}
+
+std::string concat(const std::string & prefix, const std::string & frame)
+{
+  if (prefix.empty()) {
+    return frame;
+  }
+
+  std::string composite = prefix;
+  composite.append("/");
+  composite.append(frame);
+  return composite;
+}
+
+void EffortDisplay::processMessage(sensor_msgs::msg::JointState::ConstSharedPtr msg)
+{
+  // Robot model might not be loaded already
+  if (!robot_model_) {
+    std::cerr << "Robot model might not be loaded already" << '\n';
```

Comment:
```suggestion
  // Robot model might not be loaded yet
  if (!robot_model_) {
    std::cerr << "Robot model might not be loaded yet" << '\n';
```

Besides this, an unconditional print here seems problematic.  It seems like 1) the user might miss this on the console, and 2) we might spam the user early on with a lot of these messages.  Could we instead update the Status field somehow?